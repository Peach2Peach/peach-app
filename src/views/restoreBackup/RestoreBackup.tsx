import React, { ReactElement, useContext, useEffect, useState } from 'react'

import { account } from '../../utils/account'
import { StackNavigationProp } from '@react-navigation/stack'
import i18n from '../../utils/i18n'
import LanguageContext from '../../contexts/language'
import { MessageContext } from '../../contexts/message'
import AutoScan from './AutoScan'
import Manual from './Manual'
import Restored from './Restored'
import { View } from 'react-native'
import tw from '../../styles/tailwind'


type ProfileScreenNavigationProp = StackNavigationProp<RootStackParamList, 'restoreBackup'>
type Props = {
  navigation: ProfileScreenNavigationProp;
}

// TODO
// 1. scan phone, gdrive, icloud for backup file
// (2. if no file found, ask for manual selection)
// 3. ask for password
// 4. backup restored > continue button
export default ({ navigation }: Props): ReactElement => {
  const [autoScanComplete, setAutoScanComplete] = useState(true)
  const [recoveredAccount, setRecoveredAccount] = useState(account)
  useContext(LanguageContext)
  const [, updateMessage] = useContext(MessageContext)

  useEffect(() => {
    setTimeout(() => {
      // TODO auto scan function
      setAutoScanComplete(true)
    }, 4000)
  }, [])


  const onError = () => {
    updateMessage({
      msg: i18n('form.password.invalid'),
      level: 'ERROR',
    })
  }

  return <View style={tw`px-6`}>
    {!autoScanComplete
      ? <AutoScan />
      : !recoveredAccount.publicKey
        ? <Manual
          navigation={navigation}
          onSuccess={(acc: Account) => setRecoveredAccount(acc)}
          onError={onError}
        />
        : <Restored navigation={navigation} />
    }
  </View>
}