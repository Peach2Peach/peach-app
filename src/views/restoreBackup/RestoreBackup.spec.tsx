import { render } from 'test-utils'
import { RestoreBackup } from './RestoreBackup'

const useRouteMock = jest.fn(() => ({ params: {} }))
jest.mock('../../hooks/useRoute', () => ({
  useRoute: () => useRouteMock(),
}))

describe('RestoreBackup', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<RestoreBackup />)
    expect(toJSON()).toMatchSnapshot()
  })
})
