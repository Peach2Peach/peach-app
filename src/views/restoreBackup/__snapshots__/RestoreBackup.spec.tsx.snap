// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestoreBackup should render correctly 1`] = `
<View
  style={
    {
      "height": "100%",
    }
  }
>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "flexShrink": 1,
              "paddingLeft": 8,
              "paddingRight": 8,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              [
                {
                  "color": "#2B1911",
                  "fontFamily": "Baloo2-Regular",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 24,
                },
                [
                  {
                    "fontFamily": "Baloo2-Bold",
                    "fontSize": 16,
                    "letterSpacing": 0.16,
                    "lineHeight": 20,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#FFFCFA",
                  },
                  false,
                ],
              ]
            }
          >
            file backup
          </Text>
        </View>
        <View
          style={
            [
              {
                "height": 2,
                "width": "100%",
              },
              {
                "backgroundColor": "#FFFCFA",
              },
            ]
          }
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "flexShrink": 1,
              "paddingLeft": 8,
              "paddingRight": 8,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              [
                {
                  "color": "#2B1911",
                  "fontFamily": "Baloo2-Regular",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 24,
                },
                [
                  {
                    "fontFamily": "Baloo2-Bold",
                    "fontSize": 16,
                    "letterSpacing": 0.16,
                    "lineHeight": 20,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#FCCCB6",
                  },
                  false,
                ],
              ]
            }
          >
            seed phrase
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "column",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
          {
            "flexShrink": 1,
            "height": "100%",
          },
        ]
      }
    >
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "column",
            "flexShrink": 1,
            "height": "100%",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexShrink": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              [
                {
                  "color": "#2B1911",
                  "fontFamily": "Baloo2-Regular",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 24,
                },
                {
                  "color": "#FFFCFA",
                  "fontFamily": "Baloo2-SemiBold",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 28,
                  "marginTop": 16,
                  "textAlign": "center",
                },
              ]
            }
          >
            select your backup file & enter yourÂ password
          </Text>
          <View
            style={
              {
                "marginTop": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "width": "100%",
              }
            }
          >
            <View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "gap": 4,
                      "justifyContent": "space-between",
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "width": "100%",
                    },
                    {
                      "borderRadius": 12,
                      "borderWidth": 1,
                      "overflow": "hidden",
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "borderColor": "#FFFCFA",
                    },
                    false,
                    false,
                    {
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={false}
                  autoCapitalize="none"
                  autoCorrect={false}
                  blurOnSubmit={false}
                  editable={false}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onEndEditing={[Function]}
                  onFocus={[Function]}
                  onPressIn={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="backup file"
                  placeholderTextColor="#FCCCB6"
                  removeClippedSubviews={false}
                  style={
                    [
                      {
                        "flexShrink": 1,
                        "fontFamily": "Baloo2-Regular",
                        "fontSize": 18,
                        "height": 40,
                        "letterSpacing": 0.18,
                        "lineHeight": 29.25,
                        "paddingBottom": 0,
                        "paddingTop": 0,
                        "width": "100%",
                      },
                      {
                        "color": "#FCCCB6",
                      },
                      false,
                      {
                        "borderColor": "transparent",
                        "borderWidth": 1,
                      },
                      false,
                      false,
                    ]
                  }
                  textAlignVertical="center"
                  value=""
                />
                <View
                  style={
                    {
                      "display": "flex",
                      "flexDirection": "row",
                      "gap": 16,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <test-file-stub
                      fill="#FFFCFA"
                      style={
                        [
                          {
                            "height": 24,
                            "width": 24,
                          },
                          {
                            "height": 20,
                            "width": 20,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <Text
                allowFontScaling={false}
                style={
                  [
                    {
                      "color": "#2B1911",
                      "fontFamily": "Baloo2-Regular",
                      "fontSize": 16,
                      "letterSpacing": 0.16,
                      "lineHeight": 24,
                    },
                    [
                      {
                        "fontFamily": "Baloo2-Medium",
                        "fontSize": 14,
                        "letterSpacing": 0.14,
                        "lineHeight": 20,
                        "marginLeft": 12,
                        "marginTop": 4,
                      },
                      {
                        "color": "#FFFCFA",
                      },
                    ],
                  ]
                }
              >
                 
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "gap": 4,
                      "justifyContent": "space-between",
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "width": "100%",
                    },
                    {
                      "borderRadius": 12,
                      "borderWidth": 1,
                      "overflow": "hidden",
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "borderColor": "#FFFCFA",
                    },
                    false,
                    false,
                    {
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={false}
                  autoCapitalize="none"
                  autoCorrect={false}
                  blurOnSubmit={false}
                  editable={true}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onEndEditing={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="your password"
                  placeholderTextColor="#FCCCB6"
                  removeClippedSubviews={false}
                  secureTextEntry={true}
                  style={
                    [
                      {
                        "flexShrink": 1,
                        "fontFamily": "Baloo2-Regular",
                        "fontSize": 18,
                        "height": 40,
                        "letterSpacing": 0.18,
                        "lineHeight": 29.25,
                        "paddingBottom": 0,
                        "paddingTop": 0,
                        "width": "100%",
                      },
                      {
                        "color": "#FCCCB6",
                      },
                      false,
                      {
                        "borderColor": "transparent",
                        "borderWidth": 1,
                      },
                      false,
                      false,
                    ]
                  }
                  textAlignVertical="center"
                  value=""
                />
                <View
                  style={
                    {
                      "display": "flex",
                      "flexDirection": "row",
                      "gap": 16,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <test-file-stub
                      fill="#FFFCFA"
                      style={
                        [
                          {
                            "height": 24,
                            "width": 24,
                          },
                          {
                            "height": 20,
                            "width": 20,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <Text
                allowFontScaling={false}
                style={
                  [
                    {
                      "color": "#2B1911",
                      "fontFamily": "Baloo2-Regular",
                      "fontSize": 16,
                      "letterSpacing": 0.16,
                      "lineHeight": 24,
                    },
                    [
                      {
                        "fontFamily": "Baloo2-Medium",
                        "fontSize": 14,
                        "letterSpacing": 0.14,
                        "lineHeight": 20,
                        "marginLeft": 12,
                        "marginTop": 4,
                      },
                      {
                        "color": "#FFFCFA",
                      },
                    ],
                  ]
                }
              >
                 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "paddingBottom": 32,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FCCCB6",
                "borderRadius": 9999,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "opacity": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
          >
            <Text
              allowFontScaling={false}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#2B1911",
                    "fontFamily": "Baloo2-Regular",
                    "fontSize": 16,
                    "letterSpacing": 0.16,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "color": "#F56522",
                    },
                    {
                      "fontFamily": "Baloo2-SemiBold",
                      "fontSize": 14,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "minWidth": 96,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    },
                  ],
                ]
              }
            >
              restore backup
            </Text>
            <test-file-stub
              fill="#F56522"
              style={
                [
                  {
                    "height": 24,
                    "width": 24,
                  },
                  {
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
